# Raw Data Analysis
In this notebook I explore the three data files collected from SWITRS. The purpose of this notebooks is to leek through each of the datasets and select the columns that will be used for the rest of the project. At the end of this notebook I hope to export a dataframe that I will clean and explore in a seperate notebook. While we are not making the final dataframe used in our model, this initial exploration will allow me to filter out data that is irrelevant to my project.

### Import Data

In [1]:
# import libraries
import pandas as pd
import numpy as np

In [2]:
collisions = pd.read_csv('CollisionRecords.csv', engine='python', encoding='latin1', on_bad_lines='skip')


In [3]:
party = pd.read_csv('PartyRecords.csv')

C:\Users\mende\AppData\Local\Temp\ipykernel_11452\3934567607.py:1: DtypeWarning: Columns (2,28) have mixed types. Specify dtype option on import or set low_memory=False.
  party = pd.read_csv('PartyRecords.csv')


In [4]:
victims = pd.read_csv('VictimRecords.csv')

C:\Users\mende\AppData\Local\Temp\ipykernel_11452\3289804441.py:1: DtypeWarning: Columns (6,9) have mixed types. Specify dtype option on import or set low_memory=False.
  victims = pd.read_csv('VictimRecords.csv')


## Raw Data At A Glance

In [12]:
print("Collisions Shape:", collisions.shape)
print("Victims Shape:", victims.shape)
print("Party Shape:", party.shape)

Collisions Shape: (2486655, 76)
Victims Shape: (1762334, 10)
Party Shape: (4911120, 33)


### Party Data

In [8]:
party.sample(10)

CASE_ID  PARTY_NUMBER PARTY_TYPE AT_FAULT PARTY_SEX  PARTY_AGE  \
4417163   9372859             1          1        Y         -        998   
2950859  91470858             1          1        Y         M         38   
372247   82262674             1          1        Y         F         29   
581022    8649679             2          1        N         F         52   
4520802   9433810             2          1        N         M         29   
4464555   9403358             1          1        Y         F         40   
4859274   9616713             1          1        Y         M         64   
694271    8708330             2          1        N         M         21   
1374685   9060528             2          1        N         F         21   
2061847  90978928             2          1        N         F         24   

        PARTY_SOBRIETY PARTY_DRUG_PHYSICAL DIR_OF_TRAVEL PARTY_SAFETY_EQUIP_1  \
4417163              G                   G             S                    -   
2950859              A                   -             N                    M   
372247               A                   -             W                    L   
581022               A                   -             S                    M   
4520802              H                   H             N                    M   
4464555              A                   -             N                    L   
4859274              A                   -             W                    M   
694271               A                   -             W                    L   
1374685              H                   H             N                    M   
2061847              A                   -             E                    M   

         ... MOVE_PRE_ACC VEHICLE_YEAR VEHICLE_MAKE STWD_VEHICLE_TYPE  \
4417163  ...            D          NaN            -                 -   
2950859  ...            J       2006.0        MAZDA                 A   
372247   ...            F       2021.0          KIA                 A   
581022   ...            A       2010.0       TOYOTA                 A   
4520802  ...            A       2013.0            -                 A   
4464555  ...            B       2015.0            -                 A   
4859274  ...            B       2010.0            -                 A   
694271   ...            A       2000.0        HONDA                 A   
1374685  ...            B       2018.0       NISSAN                 A   
2061847  ...            B       2020.0         TOYT                 A   

        CHP_VEH_TYPE_TOWING CHP_VEH_TYPE_TOWED RACE  INATTENTION  \
4417163                99.0                NaN  NaN          NaN   
2950859                 1.0                NaN    H            K   
372247                  1.0                NaN    H          NaN   
581022                  1.0                NaN    W          NaN   
4520802                 1.0                NaN    A          NaN   
4464555                 1.0                NaN  NaN          NaN   
4859274                 1.0                NaN    H          NaN   
694271                  1.0                NaN    W          NaN   
1374685                 1.0                NaN    W          NaN   
2061847                 1.0                NaN    A          NaN   

        SPECIAL_INFO_F SPECIAL_INFO_G  
4417163              -              -  
2950859              -              -  
372247               -              -  
581022               -              -  
4520802              -              -  
4464555              -              -  
4859274              -              -  
694271               -              -  
1374685              -              -  
2061847              -              -  

[10 rows x 33 columns]

In [30]:
party.columns

Index(['CASE_ID', 'PARTY_NUMBER', 'PARTY_TYPE', 'AT_FAULT', 'PARTY_SEX',
       'PARTY_AGE', 'PARTY_SOBRIETY', 'PARTY_DRUG_PHYSICAL', 'DIR_OF_TRAVEL',
       'PARTY_SAFETY_EQUIP_1', 'PARTY_SAFETY_EQUIP_2', 'FINAN_RESPONS',
       'SP_INFO_1', 'SP_INFO_2', 'SP_INFO_3', 'OAF_VIOLATION_CODE',
       'OAF_VIOL_CAT', 'OAF_VIOL_SECTION', 'OAF_VIOLATION_SUFFIX', 'OAF_1',
       'OAF_2', 'PARTY_NUMBER_KILLED', 'PARTY_NUMBER_INJURED', 'MOVE_PRE_ACC',
       'VEHICLE_YEAR', 'VEHICLE_MAKE', 'STWD_VEHICLE_TYPE',
       'CHP_VEH_TYPE_TOWING', 'CHP_VEH_TYPE_TOWED', 'RACE', 'INATTENTION',
       'SPECIAL_INFO_F', 'SPECIAL_INFO_G'],
      dtype='object')

In [36]:
party.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4911120 entries, 0 to 4911119
Data columns (total 33 columns):
 #   Column                Dtype  
---  ------                -----  
 0   CASE_ID               int64  
 1   PARTY_NUMBER          int64  
 2   PARTY_TYPE            object 
 3   AT_FAULT              object 
 4   PARTY_SEX             object 
 5   PARTY_AGE             int64  
 6   PARTY_SOBRIETY        object 
 7   PARTY_DRUG_PHYSICAL   object 
 8   DIR_OF_TRAVEL         object 
 9   PARTY_SAFETY_EQUIP_1  object 
 10  PARTY_SAFETY_EQUIP_2  object 
 11  FINAN_RESPONS         object 
 12  SP_INFO_1             object 
 13  SP_INFO_2             object 
 14  SP_INFO_3             object 
 15  OAF_VIOLATION_CODE    object 
 16  OAF_VIOL_CAT          object 
 17  OAF_VIOL_SECTION      float64
 18  OAF_VIOLATION_SUFFIX  object 
 19  OAF_1                 object 
 20  OAF_2                 object 
 21  PARTY_NUMBER_KILLED   int64  
 22  PARTY_NUMBER_INJURED  int64  
 23  MOVE_PR

In [37]:
# view missing values
party_miss_vals = party.isna().sum()

print("Party missing vals:", "\n",party_miss_vals[party_miss_vals>0])

Party missing vals: 
 PARTY_DRUG_PHYSICAL       11159
FINAN_RESPONS            294792
OAF_VIOLATION_CODE       312384
OAF_VIOL_SECTION        4598728
OAF_VIOLATION_SUFFIX    4826341
VEHICLE_YEAR             497839
VEHICLE_MAKE                 44
CHP_VEH_TYPE_TOWING      212539
CHP_VEH_TYPE_TOWED      4771117
RACE                     915274
INATTENTION             4817319
SPECIAL_INFO_G                4
dtype: int64


### Victims Data
We will not be using this data since what we need is captured in the other two dataframes.

In [5]:
victims.sample(10)

CASE_ID  PARTY_NUMBER  VICTIM_ROLE VICTIM_SEX  VICTIM_AGE  \
1624916   9492714             2            1          F          54   
594034    9020360             1            1          F          26   
331334    8752357             2            2          F          59   
889483   91121887             2            1          F          28   
1511724   9361558             2            1          F          35   
486196    8911193             2            1          F          88   
987654   91317603             1            1          M          20   
1333420  92101528             1            3          M          54   
1020735  91389053             1            1          F          44   
1597797   9463582             2            2          M          20   

         VICTIM_DEGREE_OF_INJURY VICTIM_SEATING_POSITION  \
1624916                        7                       1   
594034                         7                       1   
331334                         4                       4   
889483                         7                       1   
1511724                        4                       1   
486196                         1                       1   
987654                         7                       1   
1333420                        6                       0   
1020735                        6                       1   
1597797                        0                       3   

        VICTIM_SAFETY_EQUIP_1 VICTIM_SAFETY_EQUIP_2 VICTIM_EJECTED  
1624916                     M                     G              0  
594034                      L                     G              0  
331334                      L                     G              0  
889483                      M                     G              0  
1511724                     L                     G              0  
486196                      P                     G              0  
987654                      M                     G              0  
1333420                     P                     P              0  
1020735                     L                     G              0  
1597797                     L                     E              0

In [38]:
victims.columns

Index(['CASE_ID', 'PARTY_NUMBER', 'VICTIM_ROLE', 'VICTIM_SEX', 'VICTIM_AGE',
       'VICTIM_DEGREE_OF_INJURY', 'VICTIM_SEATING_POSITION',
       'VICTIM_SAFETY_EQUIP_1', 'VICTIM_SAFETY_EQUIP_2', 'VICTIM_EJECTED'],
      dtype='object')

In [40]:
victims.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1762334 entries, 0 to 1762333
Data columns (total 10 columns):
 #   Column                   Dtype 
---  ------                   ----- 
 0   CASE_ID                  int64 
 1   PARTY_NUMBER             int64 
 2   VICTIM_ROLE              int64 
 3   VICTIM_SEX               object
 4   VICTIM_AGE               int64 
 5   VICTIM_DEGREE_OF_INJURY  int64 
 6   VICTIM_SEATING_POSITION  object
 7   VICTIM_SAFETY_EQUIP_1    object
 8   VICTIM_SAFETY_EQUIP_2    object
 9   VICTIM_EJECTED           object
dtypes: int64(5), object(5)
memory usage: 134.5+ MB


In [41]:
# view missing values
victims_miss_vals = victims.isna().sum()

print("Victims missing vals:","\n",victims_miss_vals[victims_miss_vals>0])

Victims missing vals: 
 Series([], dtype: int64)


### Collisions Data

In [6]:
collisions.sample(10)

CASE_ID  ACCIDENT_YEAR  PROC_DATE JURIS  COLLISION_DATE  \
1910096  91996383           2023   20230216  9435        20230209   
2267641   9405194           2021   20220215  0106        20211107   
696076   90650614           2018   20180129  9650        20180122   
410754    8776833           2018   20190130  4905        20181218   
1293626  91250212           2020   20200610  9265        20200531   
927451   90879595           2018   20181210  9765        20181205   
1616204   9162457           2020   20201015  3313        20200805   
1855149  91928184           2022   20221128  9660        20221122   
2204010   9321415           2022   20220803  3703        20220701   
1373298  91335453           2020   20201030  9390        20201021   

         COLLISION_TIME OFFICER_ID REPORTING_DISTRICT  DAY_OF_WEEK  CHP_SHIFT  \
1910096            1835     021086                NaN            4          2   
2267641             845        412                NaN            7          5   
696076              925     013570                NaN            1          1   
410754             1231      SR533              10609            2          5   
1293626            1308     013515                NaN            7          1   
927451              740     016234                NaN            3          1   
1616204            1558       2029                  2            3          5   
1855149            1431     023001                NaN            2          2   
2204010               8       1392                  3            5          5   
1373298            1426     018869                NaN            3          2   

         ...  COUNT_PED_KILLED  COUNT_PED_INJURED  COUNT_BICYCLIST_KILLED  \
1910096  ...                 0                  0                       0   
2267641  ...                 0                  1                       0   
696076   ...                 0                  0                       0   
410754   ...                 0                  0                       0   
1293626  ...                 0                  0                       0   
927451   ...                 0                  0                       0   
1616204  ...                 0                  0                       0   
1855149  ...                 0                  0                       0   
2204010  ...                 0                  0                       0   
1373298  ...                 0                  0                       0   

         COUNT_BICYCLIST_INJURED COUNT_MC_KILLED COUNT_MC_INJURED  \
1910096                        0               0                0   
2267641                        0               0                0   
696076                         0               0                0   
410754                         0               0                0   
1293626                        0               0                0   
927451                         0               0                0   
1616204                        0               0                0   
1855149                        0               0                0   
2204010                        0               0                0   
1373298                        0               0                0   

         PRIMARY_RAMP SECONDARY_RAMP  LATITUDE  LONGITUDE  
1910096             -              -  36.75441  119.76919  
2267641             -              -  37.63734  122.07017  
696076              -              -  33.18312  117.29887  
410754              -              -       NaN        NaN  
1293626             -              -  37.99282  121.26596  
927451              -              -  34.22485  119.15803  
1616204             -              -       NaN        NaN  
1855149             -              -  33.61006  114.82376  
2204010             -              -  32.67700  117.17221  
1373298             -              -  37.70301  121.75593  

[10 rows x 76 columns]

In [7]:
collisions.columns

Index(['CASE_ID', 'ACCIDENT_YEAR', 'PROC_DATE', 'JURIS', 'COLLISION_DATE',
       'COLLISION_TIME', 'OFFICER_ID', 'REPORTING_DISTRICT', 'DAY_OF_WEEK',
       'CHP_SHIFT', 'POPULATION', 'CNTY_CITY_LOC', 'SPECIAL_COND', 'BEAT_TYPE',
       'CHP_BEAT_TYPE', 'CITY_DIVISION_LAPD', 'CHP_BEAT_CLASS', 'BEAT_NUMBER',
       'PRIMARY_RD', 'SECONDARY_RD', 'DISTANCE', 'DIRECTION', 'INTERSECTION',
       'WEATHER_1', 'WEATHER_2', 'STATE_HWY_IND', 'CALTRANS_COUNTY',
       'CALTRANS_DISTRICT', 'STATE_ROUTE', 'ROUTE_SUFFIX', 'POSTMILE_PREFIX',
       'POSTMILE', 'LOCATION_TYPE', 'RAMP_INTERSECTION', 'SIDE_OF_HWY',
       'TOW_AWAY', 'COLLISION_SEVERITY', 'NUMBER_KILLED', 'NUMBER_INJURED',
       'PARTY_COUNT', 'PRIMARY_COLL_FACTOR', 'PCF_CODE_OF_VIOL',
       'PCF_VIOL_CATEGORY', 'PCF_VIOLATION', 'PCF_VIOL_SUBSECTION',
       'HIT_AND_RUN', 'TYPE_OF_COLLISION', 'MVIW', 'PED_ACTION',
       'ROAD_SURFACE', 'ROAD_COND_1', 'ROAD_COND_2', 'LIGHTING',
       'CONTROL_DEVICE', 'CHP_ROAD_TYPE', 'PEDESTRIAN_

In [100]:
collisions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2486655 entries, 0 to 2486654
Data columns (total 76 columns):
 #   Column                   Dtype  
---  ------                   -----  
 0   CASE_ID                  int64  
 1   ACCIDENT_YEAR            int64  
 2   PROC_DATE                int64  
 3   JURIS                    object 
 4   COLLISION_DATE           int64  
 5   COLLISION_TIME           int64  
 6   OFFICER_ID               object 
 7   REPORTING_DISTRICT       object 
 8   DAY_OF_WEEK              int64  
 9   CHP_SHIFT                int64  
 10  POPULATION               int64  
 11  CNTY_CITY_LOC            int64  
 12  SPECIAL_COND             int64  
 13  BEAT_TYPE                int64  
 14  CHP_BEAT_TYPE            object 
 15  CITY_DIVISION_LAPD       object 
 16  CHP_BEAT_CLASS           int64  
 17  BEAT_NUMBER              object 
 18  PRIMARY_RD               object 
 19  SECONDARY_RD             object 
 20  DISTANCE                 float64
 21  DIRECTIO

In [42]:
# view missing values
collision_miss_vals = collisions.isna().sum()

print("Collision missing vals:","\n",collision_miss_vals[collision_miss_vals>0])

Collision missing vals: 
 OFFICER_ID                 1606
REPORTING_DISTRICT      1592911
CITY_DIVISION_LAPD      2372460
BEAT_NUMBER              155545
SECONDARY_RD                 70
DIRECTION                605699
STATE_HWY_IND              2734
CALTRANS_COUNTY         2422020
CALTRANS_DISTRICT       2422020
STATE_ROUTE             2422020
ROUTE_SUFFIX            2422035
POSTMILE_PREFIX         2422024
POSTMILE                2422020
LOCATION_TYPE           2422020
RAMP_INTERSECTION       2422020
SIDE_OF_HWY             2422040
TOW_AWAY                  31018
PCF_VIOLATION            144146
PCF_VIOL_SUBSECTION     1581598
PEDESTRIAN_ACCIDENT     2410387
BICYCLE_ACCIDENT        2427518
MOTORCYCLE_ACCIDENT     2398630
TRUCK_ACCIDENT          2327746
ALCOHOL_INVOLVED        2209823
CHP_VEHTYPE_AT_FAULT      95780
LATITUDE                 757214
LONGITUDE                757214
dtype: int64


### Notes:
- Relevant columns must be chosen for project
- Imported data had mixed data types that need to be changed to the appropriate data type using the SWITRS data decription
- Values must be appropriately encoded to fit data description. Ex: Age 998 = Not Stated 
- Missing values must be imputed or removed
- Final columns must be combined to make a new data set

## Select Data
For the first round of selecting data, I will manually go into the data description and choose columns that do not contribute to the purpose of the project such as `CHP Vehicle Type Towing` since this happens after the incedent and it's not something that lead to a collision. Later in the project we will use more advance techniques to filter through column selection for the model. Victims data will not be used since the information is captured in the party dataframe and other columns aren't needed.

### Collisions
List of columns:  
Case ID  
Collision Date  
Collision Time  
Reporting District  
Day of the Week  
CHP Shift  
Population  
County City Location  
Beat Type  
CHP Beat Type (A beat is a specific patrol area within a service area where a violation occurs.)  
Primary Rd    
Secondary Rd  
Direction  
Intersection  
Weather 1  
Weather 2  
State Highway Indicator    
Location Type    
Ramp Intersection    
Side of Highway    
Party Count  
Primary Collision Factor  
PCF Violation Code  
PCF Violation Category (Target)  
Type of Collision  
Motor Vehicle Involved With  
Ped Action  
Road Surface  
Road Condition 1  
Road Condition 2  
Lighting  
Control Device  
Ped Collision  
Bicycle Collision  
Motorcycle Collision  
Truck Collision  
Not Private Property  
Alcohol Involved     
Latitude  
Longitude


In [85]:
# choose columns to keep for collision
cols_to_keep = ['CASE_ID','COLLISION_DATE','COLLISION_TIME','REPORTING_DISTRICT','DAY_OF_WEEK',
                'CHP_SHIFT','POPULATION','CNTY_CITY_LOC','BEAT_TYPE','CHP_BEAT_TYPE','DISTANCE',
                'DIRECTION','INTERSECTION','WEATHER_1','WEATHER_2','STATE_HWY_IND','LOCATION_TYPE',
                'RAMP_INTERSECTION','SIDE_OF_HWY','PARTY_COUNT','PRIMARY_COLL_FACTOR','PCF_CODE_OF_VIOL',
                'PCF_VIOL_CATEGORY','TYPE_OF_COLLISION','MVIW', 'PED_ACTION','ROAD_SURFACE','ROAD_COND_1',
                'ROAD_COND_2','LIGHTING','CONTROL_DEVICE','PEDESTRIAN_ACCIDENT','BICYCLE_ACCIDENT', 
                'MOTORCYCLE_ACCIDENT','TRUCK_ACCIDENT','NOT_PRIVATE_PROPERTY','ALCOHOL_INVOLVED',
                'LATITUDE','LONGITUDE']

reduced_collisions = collisions[cols_to_keep]

In [86]:
reduced_collisions.sample(10)

CASE_ID  COLLISION_DATE  COLLISION_TIME REPORTING_DISTRICT  \
2259573   9391451        20211212            1015               0610   
700721   90655162        20180126            2252                NaN   
1431911  91403864        20210117            1750                NaN   
1260448  91213421        20200229             340                NaN   
1694179  91723261        20220311            1150                NaN   
1568796  91567126        20210816            1452                NaN   
829271   90783169        20180729             116                NaN   
1160346  91111315        20191022            1915                NaN   
1512670  91499563        20210608            1352                NaN   
279956    8638449        20180603            1603                 9B   

         DAY_OF_WEEK  CHP_SHIFT  POPULATION  CNTY_CITY_LOC  BEAT_TYPE  \
2259573            7          5           6           1985          0   
700721             5          3           5            112          1   
1431911            7          2           4           3618          1   
1260448            6          3           9           4100          1   
1694179            5          1           7           3711          1   
1568796            1          2           6           3345          1   
829271             7          3           9           3700          1   
1160346            2          2           7           3711          1   
1512670            2          1           9           1900          1   
279956             7          5           6           3315          0   

        CHP_BEAT_TYPE  ...  LIGHTING CONTROL_DEVICE PEDESTRIAN_ACCIDENT  \
2259573             0  ...         A              A                 NaN   
700721              1  ...         C              D                 NaN   
1431911             1  ...         D              D                 NaN   
1260448             3  ...         D              D                 NaN   
1694179             1  ...         A              D                 NaN   
1568796             3  ...         A              D                 NaN   
829271              1  ...         D              D                 NaN   
1160346             1  ...         C              D                 NaN   
1512670             1  ...         A              D                 NaN   
279956              0  ...         A              A                 NaN   

        BICYCLE_ACCIDENT MOTORCYCLE_ACCIDENT TRUCK_ACCIDENT  \
2259573              NaN                 NaN            NaN   
700721               NaN                 NaN            NaN   
1431911              NaN                 NaN              Y   
1260448              NaN                 NaN            NaN   
1694179              NaN                 NaN              Y   
1568796              NaN                 NaN            NaN   
829271               NaN                 NaN            NaN   
1160346              NaN                 NaN            NaN   
1512670              NaN                 NaN            NaN   
279956               NaN                 NaN            NaN   

        NOT_PRIVATE_PROPERTY ALCOHOL_INVOLVED  LATITUDE  LONGITUDE  
2259573                    Y              NaN  34.42062  118.48455  
700721                     Y              NaN  37.71966  122.17698  
1431911                    Y              NaN  34.13155  117.20811  
1260448                    Y                Y  37.50792  122.35537  
1694179                    Y              NaN  32.78389  117.11391  
1568796                    Y              NaN  34.01929  117.49056  
829271                     Y                Y  33.35462  117.15894  
1160346                    Y              NaN  32.77839  117.10736  
1512670                    Y              NaN  34.28231  118.40134  
279956                     Y              NaN  33.88548  117.53619  

[10 rows x 39 columns]

### Party 
List of Columns:  
Case id (to merge on)  
At Fault  
Party Sex  
Party Age  
Party Sobriety  
Party Drug Physical   
Direction of travel  
Party Safety Equiptment 1  
Party Safety Equiptment 2  
Financial Responsibility  
Special Info 2  (Cell use)
OAF Violation Code  
OAF Violation Category  
OAF Factor 1  
OAF Factor 2  
Movement Preceding Collision    
Vehicle Year  
Vehicle Make  
Statewide Vehicle Type  
CHP Vehicle Type  
CHP Vehicle Type Towed  
Party Race  



In [77]:
# choose columns to keep for party
party_col_to_keep = ['CASE_ID','AT_FAULT','PARTY_SEX','PARTY_AGE','PARTY_SOBRIETY','PARTY_DRUG_PHYSICAL', 
                     'DIR_OF_TRAVEL','PARTY_SAFETY_EQUIP_1','PARTY_SAFETY_EQUIP_2','FINAN_RESPONS','SP_INFO_2',
                     'OAF_VIOLATION_CODE','OAF_VIOL_CAT','OAF_1','OAF_2','MOVE_PRE_ACC','VEHICLE_YEAR','VEHICLE_MAKE',
                     'STWD_VEHICLE_TYPE','CHP_VEH_TYPE_TOWING','CHP_VEH_TYPE_TOWED','RACE']

party_reduced = party[party_col_to_keep]

In [88]:
party_reduced.sample(10)


CASE_ID AT_FAULT PARTY_SEX  PARTY_AGE PARTY_SOBRIETY  \
2234928  91066468        N         M         23              A   
3497494  91821743        N         F         43              A   
3552467  91856988        N         M         34              A   
258373   82013237        Y         M        998              G   
1897982  90895621        Y         M         92              A   
3431105  91778141        N         -        998              G   
2301335  91099417        N         M         46              A   
2622963  91273944        N         M         69              A   
19617    81330620        N         M         23              A   
4236886   9260882        N         M        998              A   

        PARTY_DRUG_PHYSICAL DIR_OF_TRAVEL PARTY_SAFETY_EQUIP_1  \
2234928                   -             S                    M   
3497494                   -             N                    M   
3552467                   -             N                    M   
258373                    G             W                    -   
1897982                   -             N                    M   
3431105                   G             N                    B   
2301335                   -             S                    P   
2622963                   -             E                    P   
19617                     -             E                    M   
4236886                   -             W                    -   

        PARTY_SAFETY_EQUIP_2 FINAN_RESPONS  ... OAF_VIOL_CAT OAF_1 OAF_2  \
2234928                    G             Y  ...           -      N     -   
3497494                    G             Y  ...           -      N     -   
3552467                    G             Y  ...           -      N     -   
258373                     -             N  ...           -      N     -   
1897982                    G             Y  ...           -      N     -   
3431105                    B             N  ...           -      N     -   
2301335                    G             Y  ...           -      N     -   
2622963                    G             Y  ...           -      N     -   
19617                      G             Y  ...           -      N     -   
4236886                    -             N  ...           -      N     -   

        MOVE_PRE_ACC VEHICLE_YEAR VEHICLE_MAKE  STWD_VEHICLE_TYPE  \
2234928            A       2012.0         MITS                  A   
3497494            H       2012.0         TOYT                  A   
3552467            A       2022.0         MAZD                  A   
258373             G       1997.0    CHEVROLET                  D   
1897982            D       2016.0         CHEV                  D   
3431105            J          NaN      UNKNOWN                  -   
2301335            H       2015.0          FRH                  G   
2622963            B       1997.0        DODGE                  D   
19617              D       1997.0     PLYMOUTH                  A   
4236886            B          NaN            -                  -   

        CHP_VEH_TYPE_TOWING CHP_VEH_TYPE_TOWED  RACE  
2234928                 1.0                NaN     H  
3497494                 1.0                NaN     H  
3552467                 7.0                NaN     O  
258373                 22.0                NaN     B  
1897982                22.0                NaN     W  
3431105                99.0                NaN   NaN  
2301335                25.0               31.0     O  
2622963                22.0                NaN     H  
19617                   1.0                NaN     B  
4236886                 NaN                NaN   NaN  

[10 rows x 22 columns]

### Merge Dataframes

In [91]:
accidents = party_reduced.merge(reduced_collisions, how = 'left', on = 'CASE_ID')

In [95]:
pd.set_option('display.max_columns', None)

accidents.head(10)

CASE_ID AT_FAULT PARTY_SEX  PARTY_AGE PARTY_SOBRIETY PARTY_DRUG_PHYSICAL  \
0    81715        Y         M         35              H                   H   
1    81715        N         F         43              H                   H   
2   682021        Y         F         21              C                   -   
3   682021        N         F         58              C                   -   
4   726202        Y         -        998              G                   G   
5  6698641        Y         M         52              A                   -   
6  6698641        N         F         52              A                   -   
7  6698641        N         F         54              A                   -   
8  6698645        Y         M         31              B                   -   
9  6698645        N         M         44              A                   -   

  DIR_OF_TRAVEL PARTY_SAFETY_EQUIP_1 PARTY_SAFETY_EQUIP_2 FINAN_RESPONS  \
0             N                    L                    G             Y   
1             N                    M                    G             Y   
2             W                    M                    G             Y   
3             W                    L                    G             Y   
4             N                    -                    -           NaN   
5             E                    N                    C             N   
6             E                    M                    G             Y   
7             E                    M                    G             Y   
8             N                    M                    G             N   
9             N                    M                    G             Y   

  SP_INFO_2 OAF_VIOLATION_CODE OAF_VIOL_CAT OAF_1 OAF_2 MOVE_PRE_ACC  \
0         3                  -           -      N     -            B   
1         3                  -           -      N     -            B   
2         -                  -           -      E     -            J   
3         -                  -           -      E     -            B   
4         -                  -           -      N     -            B   
5         3                  -           -      N     -            B   
6         3                  -           -      N     -            A   
7         3                  -           -      N     -            A   
8         3                NaN           25     A     -            H   
9         3                  -           -      N     -            A   

   VEHICLE_YEAR VEHICLE_MAKE STWD_VEHICLE_TYPE  CHP_VEH_TYPE_TOWING  \
0        2007.0         FORD                 -                  NaN   
1        2019.0            -                 -                  NaN   
2        2014.0            -                 A                  1.0   
3        2018.0            -                 A                  1.0   
4        2005.0            -                 A                  1.0   
5        1981.0    CHEVROLET                 D                 22.0   
6        2015.0        MAZDA                 A                  7.0   
7        2007.0    CHEVROLET                 D                 22.0   
8        1998.0       NISSAN                 A                  1.0   
9        2018.0         AUDI                 A                  1.0   

  CHP_VEH_TYPE_TOWED RACE  COLLISION_DATE  COLLISION_TIME REPORTING_DISTRICT  \
0                NaN    O      20200314.0           745.0                212   
1                NaN    H      20200314.0           745.0                212   
2                NaN    H      20210608.0           855.0                NaN   
3                NaN    A      20210608.0           855.0                NaN   
4                NaN  NaN      20200726.0           250.0                064   
5                NaN    H      20180911.0          1530.0                 27   
6                NaN    W      20180911.0          1530.0                 27   
7                NaN    H      20180911.0          1530.0                 27   
8           

In [94]:
# export data
accidents.to_csv('accident_data.csv')

This is the end of this workbook. In order to save memory on my device I will continue exploring and cleaning the data in another workbook. This will also maintain a cleaner look for the project.  
Something that I am wondering at this point of the project is if I should only consider the data for people who were at fault to avoid doubling values for some columns. 
